paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java

```diff
@@ -1167,4 +1174,15 @@ public void setBodyYaw(final float bodyYaw) {
     public boolean canUseEquipmentSlot(org.bukkit.inventory.EquipmentSlot slot) {
         return this.getHandle().canUseSlot(org.bukkit.craftbukkit.CraftEquipmentSlot.getNMS(slot));
     }
+
+    @Override
+    public CombatTracker getCombatTracker() {
+        return this.combatTracker;
+    }
+
+    @Override
+    public FallLocationType calculateFallLocationType() {
```

Comment:
I am a bit confused on these.
The entry yields this nullable, which mirrors internals, while this falls back to GENERIC.
What is the line of thought here?

Reply:
I did what vanilla does when calculating the fall message:
```java
FallLocation fallLocation = Objects.requireNonNullElse(combatEntry.fallLocation(), FallLocation.GENERIC);
```
The reasoning is, since this calculates the fall location type, then if there's no special fall type, yield `GENERIC` (generic is only used internally as a fallback when the type is `null` iirc, so it was `null` vs `GENERIC`, and I liked the latter more)
But if I've missed something, nullable is also fine

Reply:
Yea I know it is used internally for the death message computation, but that seems to be about it in terms of usage for that type?
The disconnect between "LivingEntity#calculateFallLocationType" and "create new combat entry from living entity -> ah it returns null here now eventho it just yielded GENERIC" is a bit unfortunate imo.
Tho I guess that is more on mojang than on this API given they do the "fallback" so late in their logic.
It is fine like this then, tho maybe a bit of javadocs if you have some time.

Reply:
> it returns null here now eventho it just yielded GENERIC"

It's quite ambiguous (javadoc suggestions?), but is expected from my point of view
Having `GENERIC` in the entry represents a fall (and this method is meant for that), while `null` means that the entry is not a fall damage one

Reply:
Yea but `calculateFallLocationType` yielding GENERIC then, when the entity isn't falling rn is kind of a counterpoint.
Having a non null falltype in the entry (at least according to internals) does not define if that was a fall.
That is tracked (correctly so) via the DamageSource.
You can be in the air and killed by an arrow, that kill won't be a registered as a fall related death even though the last entry does hold a falllocation type.

Javadocs on CombatEntry#combatEntry also currently link to the calc method, which then would not be correct right?